solidity/compile.ts

```diff
@@ -2,6 +2,15 @@ import { promises as fs } from 'fs'
 import path = require('path')
 import { compile } from 'solc'
 
+async function exists(path: string) {
+	try {
+		await fs.stat(path)
+		return true
+	} catch {
+		return false
+	}
+}
```

Comment:
*grumble* I hate exceptions as part of happy path, but I have already tried fighting NodeJS developers on this and failed, so I accept.

Reply:
I also was surprised this seems to be the recommended way